lms/djangoapps/courseware/tests/test_views.py

```diff
@@ -1720,6 +1720,85 @@ def test_activate_block_id(self):
+    """
+    Test the index view to handle vertical positions. Loads first position if inputted
+    position is non-positive or greater than number of positions available.
+    """
+
+    def setUp(self):
+        """
+        Set up initial test data
+        """
+        super(TestIndewViewWithVerticalPositions, self).setUp()
+
+        self.user = UserFactory()
+
+        # create course with 3 positions
+        self.course = CourseFactory.create()
+        self.chapter = ItemFactory.create(parent=self.course, category='chapter')
+        self.section = ItemFactory.create(parent=self.chapter, category='sequential', display_name="Sequence")
+        ItemFactory.create(parent=self.section, category='vertical', display_name="Vertical1")
+        ItemFactory.create(parent=self.section, category='vertical', display_name="Vertical2")
+        ItemFactory.create(parent=self.section, category='vertical', display_name="Vertical3")
+
+        self.client.login(username=self.user, password='test')
+        CourseEnrollmentFactory(user=self.user, course_id=self.course.id)
+
+    def _get_course_vertical_by_position(self, input_position):
+        """
+        Returns client response to input position.
+        """
+        return self.client.get(
+            reverse(
+                'courseware_position',
+                kwargs={
+                    'course_id': unicode(self.course.id),
+                    'chapter': self.chapter.url_name,
+                    'section': self.section.url_name,
+                    'position': input_position,
+                }
+            )
+        )
+
+    def _assert_correct_position(self, response, expected_position):
+        """
+        Asserts that the expected position and the position in the response are the same
+        """
+        self.assertIn('data-position="{}"'.format(expected_position), response.content)
+
+    def test_negative_position(self):
+        """
+        Load first position when negative position inputted
+        """
+        resp = self._get_course_vertical_by_position(-1)
+        self._assert_correct_position(resp, 1)
+
+    def test_zero_position(self):
+        """
+        Load first position when 0/-0 position inputted
+        """
+        resp = self._get_course_vertical_by_position("0")
+        self._assert_correct_position(resp, 1)
+
+        resp = self._get_course_vertical_by_position("-0")
+        self._assert_correct_position(resp, 1)
+
+    def test_positive_position(self):
+        """
+        Load given position when 0 < input_position <= num_positions_available
+        """
+        resp = self._get_course_vertical_by_position(2)
+        self._assert_correct_position(resp, 2)
+
+    def test_large_positive_position(self):
+        """
+        Load first position when positive positive > num_positions_available
+        """
+        resp = self._get_course_vertical_by_position(5)
+        self._assert_correct_position(resp, 1)
```

Comment:
It might be worth combining all these tests into a single test method, then using `ddt` to run them with a variety of `(input, expected)` tuples.

Although I guess the argument could also be made that having the separate docstrings provides some value, so whichever way you decide to go is fine with me. @jcdyer How do you feel about combined tests versus separate?